core/src/main/java/org/elasticsearch/search/rescore/QueryRescorer.java

```diff
@@ -212,7 +203,7 @@ public void setScoreMode(String scoreMode) {
     }
 
     @Override
-    public void extractTerms(SearchContext context, RescoreSearchContext rescoreContext, Set<Term> termsSet) {
+    public void extractTerms(SearchContext context, RescoreContext rescoreContext, Set<Term> termsSet) {
```

Comment:
Maybe replace `SearchContext` with `IndexSearcher` if we don't want to expose `SearchContext` in plugins.